apps/dgst.c

```diff
@@ -319,6 +319,8 @@ int dgst_main(int argc, char **argv)
         }
 
         sigkey = app_keygen(mac_ctx, mac_name, 0, 0 /* not verbose */);
+        if (sigkey == NULL)
+            goto end;
         /* Verbose output would make external-tests gost-engine fail */
         EVP_PKEY_CTX_free(mac_ctx);
```

Comment:
Do we need to free `mac_ctx` is `sigkey == NULL`?

Reply:
yes indeed, good catch.

Reply:
BTW: i am not sure if the `sigkey` might be overwritten here below
https://github.com/openssl/openssl/blob/4ee8c1fb51687ea811fc2abf87e173c70d018bc2/apps/dgst.c#L326-L333

Reply:
It cannot - see the check at line 275